cli/lib/util.js

```diff
@@ -255,31 +270,39 @@ const util = {
     return path.join(process.cwd(), '..', '..', filename)
   },
 
-  getEnv (varName) {
+  getEnv (varName, trim) {
+    la(is.unemptyString(varName), 'expected environment variable name, not', varName)
+
     const envVar = process.env[varName]
     const configVar = process.env[`npm_config_${varName}`]
     const packageConfigVar = process.env[`npm_package_config_${varName}`]
 
+    let result
+
     if (envVar) {
       debug(`Using ${varName} from environment variable`)
 
-      return envVar
-    }
-
-    if (configVar) {
+      result = envVar
+    } else if (configVar) {
       debug(`Using ${varName} from npm config`)
 
-      return configVar
-    }
-
-    if (packageConfigVar) {
+      result = configVar
+    } else if (packageConfigVar) {
       debug(`Using ${varName} from package.json config`)
 
-      return packageConfigVar
+      result = packageConfigVar
     }
 
-    return undefined
-
+    // environment variables are often set double quotes to escape characters
+    // and on Windows it can lead to weird things: for example
+    //  set FOO="C:\foo.txt" && node -e "console.log('>>>%s<<<', process.env.FOO)"
+    // will print
+    //    >>>"C:\foo.txt" <<<
+    // see https://github.com/cypress-io/cypress/issues/4506#issuecomment-506029942
+    // so for sanity sake we should first trim whitespace characters and remove
+    // double quotes around environment strings if the caller is expected to
+    // use this environment string as a file path
+    return trim ? dequote(_.trim(result)) : result
```

Comment:
Does it remove the quotes first or trim first? Does an example like "foo.txt  " work properly in your test?

Reply:
first trims the whitespace, then removes quotes, and there are unit tests, let me add one specifically for your case